common/filetoolkitwithundo.cpp

```diff
@@ -0,0 +1,204 @@
+#include "filetoolkitwithundo.h"
+
+#include "directorycleanuphandler.h"
+#include "filecleanuphandler.h"
+#include "mountcleanuphandler.h"
+
+namespace softwarecontainer {
+
+FileToolkitWithUndo::~FileToolkitWithUndo()
+{
+    bool success = true;
+    // Clean up all created directories, files, and mount points
+    for (auto it = m_cleanupHandlers.rbegin(); it != m_cleanupHandlers.rend(); ++it) {
+        // Cleaning functions will do their own error/warning output
+        if(isError((*it)->clean())) {
+            success = false;
+        }
+        delete *it;
+    }
+
+    if(!success) {
+        log_error() << "One or more cleanup handlers returned error status, please check the log";
+    }
+}
+
+ReturnCode FileToolkitWithUndo::createParentDirectory(const std::string &path)
+{
+    log_debug() << "Creating parent directories for " << path;
+    std::string parent = parentPath(path);
+    if (!isDirectory(parent) && parent != "") {
```

Comment:
I know this is just code that has been moved, but shouldn't we do && parent.empty() instead of comparing with empty string?